src/openrct2/ride/Vehicle.cpp

```diff
@@ -9890,3 +9899,150 @@ void vehicle_invalidate_window(rct_vehicle * vehicle)
+            {
+                if (!playedClaxon && 0 == (tileElement->flags & TILE_ELEMENT_FLAG_BLOCKED_BY_VEHICLE))
+                {
+                    vehicle_claxon(vehicle);
+                }
+                crossingBonus = 4;
+                tileElement->flags |= TILE_ELEMENT_FLAG_BLOCKED_BY_VEHICLE;
+            }
+            else
+            {
+                crossingBonus = 0;
+            }
+
+            if (--autoReserveAhead + crossingBonus <= 0)
+            {
+                break;
+            }
+
+            z = xyElement.element->base_height;
+
+            if (travellingForwards)
+            {
+                if (!track_block_get_next(&xyElement, &xyElement, &z, &direction))
+                {
+                    break;
+                }
+            }
+            else
+            {
+                if (!track_block_get_previous(xyElement.x, xyElement.y, xyElement.element, &output))
+                {
+                    break;
+                }
+                xyElement.x = output.begin_x;
+                xyElement.y = output.begin_y;
+                xyElement.element = output.begin_element;
+            }
+
+            if (xyElement.element->properties.track.type == TRACK_ELEM_BEGIN_STATION ||
+                xyElement.element->properties.track.type == TRACK_ELEM_MIDDLE_STATION ||
+                xyElement.element->properties.track.type == TRACK_ELEM_END_STATION)
+            {
+                break;
+            }
+        }
+    }
+
+    xyElement.x = backVehicle->track_x;
+    xyElement.y = backVehicle->track_y;
+    z           = backVehicle->track_z;
+    xyElement.element = map_get_track_element_at_of_type_seq(
+        backVehicle->track_x, backVehicle->track_y, backVehicle->track_z >> 3,
+        backVehicle->track_type >> 2, 0
+        );
+
+    if (xyElement.element)
+    {
+        uint8 freeCount = travellingForwards? 3 : 1;
+
+        while (freeCount-- > 0)
+        {
+            if (travellingForwards)
+            {
+                if (track_block_get_previous(xyElement.x, xyElement.y, xyElement.element, &output))
+                {
+                    xyElement.x = output.begin_x;
+                    xyElement.y = output.begin_y;
+                    xyElement.element = output.begin_element;
+                }
+            }
+        
+            rct_tile_element *tileElement = map_get_path_element_at(
+                xyElement.x / 32,
+                xyElement.y / 32,
+                xyElement.element->base_height
+            );
+            if (tileElement)
+            {
+                tileElement->flags &= ~TILE_ELEMENT_FLAG_BLOCKED_BY_VEHICLE;
+            }
+        }
+    }
+}
+
+void vehicle_claxon(rct_vehicle * vehicle)
+{
+    rct_ride_entry* rideEntry = get_ride_entry(vehicle->ride_subtype);
+    audio_play_sound_at_location((rideEntry->vehicles[0].sound_range == 3)? SOUND_TRAIN_WHISTLE : SOUND_TRAM, vehicle->x, vehicle->y, vehicle->z);
```

Comment:
Most vehicle sound effects are performed by the vehicle sound code rather than played directly. I'm not sure if that is the approach that should be done here but if it is I suggest using the sound2_flags (a param that is labelled in my PR).

Reply:
I did not know about the flags while writing this - letting the flags handle it sounds much more consistent. The function was mostly meant as a general way to just get a sound played for any reason.

Reply:
It's okay i only found them yesterday :)